src/access_nri_intake/experiment/utils.py

```diff
@@ -0,0 +1,326 @@
+        # If the previous check passes, then we need to check that the name in
+        # the catalog_file matches the name of the csv file. Someone might have
+        # manually fiddled with it, so we'll convert it to a path object and check
+        # the name attribute.
+        if Path(ds_json["catalog_file"]).name != self.csv_handle.name:
+            self.valid = False
+            self.invalid_ds_cause = (
+                "catalog_filename in JSON does not match csv.gz filename"
+            )
+            return None
+
+        if set(colnames) != set(
+            [item["column_name"] for item in ds_json["attributes"]]
+        ).union({"path"}):
+            self.valid = False
+            self.invalid_ds_cause = "columns specified in JSON do not match csv.gz file"
+            return None
+
+        # If all of these pass, then we can try to open the datastore
+
+    def __bool__(self):
+        return not all(
+            [
+                self.json_handle == "",
+                self.csv_handle == "",
+                not self.valid,
+                self.invalid_ds_cause == "",
+            ]
+        )
+
+
+def verify_ds_current(
+    ds_info: DatastoreInfo,
+    experiment_files: set[Path],
+) -> bool:
+    """
+    Check that the datastore is current - do we have assets in our directory that
+    are not in the datastore? Do we have assets in the datastore that are not in
+    our directory? Are the assets in the datastore the same as the assets in our
+    directory? If any of these are true, then we need to rebuild the datastore.
+
+
+
+    Parameters
+    ----------
+    ds_info : DatastoreInfo
+        The datastore information object.
+    experiment_files : set[Path]
+        The set of files found in the experiment directory. These are typically going
+        to be generated by the find_experiment_files function.
+
+    Returns
+    -------
+    bool
+        Whether the datastore is valid and up to date.
+
+    """
+
+    hashfile = (
+        Path(ds_info.json_handle).parent / f".{Path(ds_info.json_handle).stem}.hash"
+    )
+
+    if not hashfile.exists():
+        warnings.warn(
+            f"{Fore.YELLOW}No hash file found for datastore. Regenerating datastore...{Style.RESET_ALL}",
+            category=DataStoreWarning,
+            stacklevel=2,
+        )
+        return False
+
+    mf = Manifest(str(hashfile)).load()
+    manifest_files = {v.get("fullpath") for v in mf.data.values()}
+
+    # Convert experiment files to strings for compatibility with yamanifest
+    experiment_files_str = {str(file) for file in experiment_files}
+
+    if experiment_files_str != manifest_files:
+        warn_str = (
+            "extra files in"
+            if len(experiment_files_str) < len(manifest_files)
+            else "missing files from"
+        )
+        warnings.warn(
+            f"{Fore.YELLOW}Experiment directory and datastore do not match ({warn_str} datastore). Regenerating datastore...{Style.RESET_ALL}",
```

Comment:
If I'm reading this right, `verify_ds_current` isn't actually regenerating the datastore, but signalling that it needs to be regenerated. I'd recommend changing the end of the message to read "Datastore regeneration required..." or similar.